### PR TITLE
Feature/add new sleep data

### PR DESCRIPTION
**Pull request by saraho1123**
## Is this a fix or a feature?
feature/fix

## What is the change?
This branch originally started with using POST and GET to allow users to submit new data for Sleep, Hydration and Activity. This piece took longer to do than other parts and so some of the POST and GET functionality is in the previous PR. Now it is all cleaned up and looking good!

We will end up with a few merge conflicts for this PR, but I don't think the will be too difficult to resolve. 

## What does it fix?
clean up the code and make some improvements on what was in the last PR

## Where should the reviewer start?
line 30 of fetch.js
line 47 of scripts.js
line 41 of Sleep.js
line 49 of index.js

## How should this be tested?
All inputs and buttons on the site should be working!

## Relevant Ticket(s)
#12 
#13 
#14

@Shawntru 
@adavidson02
@sarah01123

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -154,6 +261,7 @@ function makeSleepQualityHTML(id, sleepInfo, userStorage, method) {
 }
 
 function addActivityInfo(id, activityInfo, dateString, userStorage, laterDateString, user, winnerId) {
+  // addDailyActivityInfo(id, activityInfo, dateString, userStorage);
```
Comment (ADavidson02) [inline src/scripts.js:264]: Do we need this commented out?

File: src/scripts.js [hunk 2]
```diff
@@ -126,13 +217,24 @@ function makeRandomDate(userStorage, id, dataSet) {
   return sortedArray[Math.floor(Math.random() * sortedArray.length + 1)].date;
 }
 
+function clearHtml(queryElements) {
+  queryElements.forEach(query => {
+    query.innerHTML = '';
+  })
+}
```
Comment (ADavidson02) [inline src/scripts.js:224]: This looks great!

File: src/Sleep.js [hunk 3]
```diff
@@ -38,7 +36,7 @@ class Sleep {
       sumSoFar += dataItem.sleepQuality;
       return sumSoFar;
     }, 0);
-    return totalSleepQuality / sleepData.length;
+    return totalSleepQuality / this.sleepData.length;
```
Comment (saraho1123) [inline src/Sleep.js:39]: There will be a merge conflict because of this little change. Sorry about that.

**Review by ADavidson02 — APPROVED** [review]
Looks great Sarah, thank you so much for all of your hard work on this branch.

File: src/scripts.js (hunk 1 above)
Comment (saraho1123) [reply src/scripts.js:264]: Oh! This should be deleted! I apologize.

File: src/fetch.js [hunk 4]
```diff
@@ -25,7 +25,61 @@ let requests = {
       .then(response => response.json())
       .then(data => data.sleepData)
       .catch(error => console.log(error))
-  }
+  },
```
Comment (Shawntru) [inline src/fetch.js:28]: Little comma hiding out here

File: src/scripts.js (hunk 1 above)
Comment (saraho1123) [reply src/scripts.js]: Fixed!

File: src/scripts.js [hunk 5]
```diff
@@ -126,13 +217,24 @@ function makeRandomDate(userStorage, id, dataSet) {
...
+
 function addHydrationInfo(id, hydrationInfo, dateString, userStorage, laterDateString) {
-  hydrationToday.insertAdjacentHTML('afterBegin', `<p>You drank</p><p><span class="number">${hydrationInfo.calculateDailyOunces(id, dateString)}</span></p><p>oz water today.</p>`);
```
Comment (Shawntru) [inline src/scripts.js:130]: I see this was removed, did we not need it or did it get moved someplace else?

**Review by Shawntru — CHANGES_REQUESTED** [review]

File: src/fetch.js (hunk 4 above)
Comment (saraho1123) [reply src/fetch.js:28]: That one does have to be there, as it is a method on the object 'requests'. But there is another one after the last method that is optional!

File: src/scripts.js (hunk 5 above)
Comment (saraho1123) [reply src/scripts.js:130]: Yes. It got moved to addDailyOunces I believe and now addDailyOunces is call there.

**Review by Shawntru — APPROVED** [review]

**Pull request merged**